processor/spanmetricsprocessor/processor.go

```diff
@@ -420,17 +432,25 @@ func (p *processorImp) aggregateMetrics(traces pdata.Traces) {
 
 func (p *processorImp) aggregateMetricsForServiceSpans(rspans pdata.ResourceSpans, serviceName string) {
 	ilsSlice := rspans.InstrumentationLibrarySpans()
+	instrLibName := instrLibKey(instrumentationLibraryName)
 	for j := 0; j < ilsSlice.Len(); j++ {
 		ils := ilsSlice.At(j)
+
+		// if confing is set to inherit instrumentation library name, then assume from trace
+		// otherwise use default
+		if p.inheritInstrumentationLibraryName {
+			instrLibName = instrLibKey(ils.InstrumentationLibrary().Name())
+		}
+
 		spans := ils.Spans()
 		for k := 0; k < spans.Len(); k++ {
 			span := spans.At(k)
-			p.aggregateMetricsForSpan(serviceName, span, rspans.Resource().Attributes())
+			p.aggregateMetricsForSpan(serviceName, span, rspans.Resource().Attributes(), instrLibName)
 		}
 	}
 }
 
-func (p *processorImp) aggregateMetricsForSpan(serviceName string, span pdata.Span, resourceAttr pdata.AttributeMap) {
+func (p *processorImp) aggregateMetricsForSpan(serviceName string, span pdata.Span, resourceAttr pdata.AttributeMap, instrLibName instrLibKey) {
```

Comment:
how about extract out the` aggregationMeta` out as:

```go
type aggregationMeta struct {
  serviceName string
  instrlibName instrLibKey
  resourceAttr pdata.AttributeMap
}

... 
func (p *processorImp) aggregateMetricsForSpan(span pdata.Span, aggregationMeta aggregationMeta) {}
```

the current signature of this function is a bit hard to read.